engine/src/main/java/org/terasology/rendering/dag/StateChange.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+/**
+ *
+ */
+public interface StateChange<T> {
+
+    T getValue();
+}
```

Comment:
Yeah your options in interfaces are somewhat limited. 

What you might consider if there's no need to ever have a method deviate is to include [default methods](https://docs.oracle.com/javase/tutorial/java/IandI/defaultmethods.html) right in the interface. That page talks a bit about static methods in interfaces too, hmm :-)